rust/connlib/tunnel/src/client.rs

```diff
@@ -1233,13 +1233,104 @@ mod tests {
     }
 
     #[test]
-    fn remove_resources_works() {}
+    fn add_resources_update_works_to_dns() {
+        let mut client_state = ClientState::for_test();
+        client_state.add_resources(&[cidr_resource("10.0.0.0/24"), dns_resource("baz.com")]);
+        client_state.add_resources(&[cidr_resource_dns_id("11.0.0.0/24")]);
+
+        assert_eq!(
+            HashSet::<&ResourceDescription>::from_iter(client_state.resources().iter()),
+            HashSet::from_iter(
+                [
+                    cidr_resource_dns_id("11.0.0.0/24"),
+                    cidr_resource("10.0.0.0/24")
+                ]
+                .iter()
+            )
+        );
```

Comment:
Could reduce some duplication here by making a macro `assert_iter_eq!` that takes two iterators and compares them via `HashSet`s.

Reply:
I introduced the `hashset` function, which we still need to call `iter` but saves us the repetition of having to be explicit about the types by getting the `IntoIter` trait out of the way